source/signalzeug/source/ScopedConnection.cpp

```diff
@@ -14,15 +14,32 @@ ScopedConnection::ScopedConnection(const Connection & connection)
 {
 }
 
+ScopedConnection::ScopedConnection(ScopedConnection && other)
+:   m_connection(other.m_connection)
+{
+    other.m_connection = Connection();
+}
+
 ScopedConnection::~ScopedConnection()
 {
 	m_connection.disconnect();
 }
 
-void ScopedConnection::operator=(const Connection & connection)
+ScopedConnection& ScopedConnection::operator=(const Connection & connection)
 {
 	m_connection.disconnect();
 	m_connection = connection;
+
+    return *this;
+}
+
+ScopedConnection& ScopedConnection::operator=(ScopedConnection && other)
+{
+    m_connection.disconnect();
+    m_connection = other.m_connection;
```

Comment:
Actually `std::swap` is correct here. You want the old content of m_connection to be freed up by the `other` object.